lib/jirawatch/cli/track.rb

```diff
@@ -6,55 +6,84 @@ class Track < Dry::CLI::Command
       include Jirawatch::CLI::AuthenticatedCommand
 
       argument :issue_key, required: true, desc: "Issue key that you want to track time for"
-      option :started_at, aliases: ["-t"], desc: "Specify when you started working on this issue with a HH:mm format"
+      option :time_started_at, aliases: ["-t"], desc: "Specify when you started working on this issue with a HH:mm format"
       option :worklog_message, aliases: ["-m"], desc: "Specify a work log message"
 
       def call(issue_key:, **options)
         @jira_client.Issue.find(issue_key) # Fails if issue doesn't exist
 
+        time_started_at = Time.parse(options.fetch(:time_started_at, Time.now.to_s))
+        time_paused_at = 0
+        time_restarted_at = time_started_at
+        partial_time_spent = 0
+
         puts "Logging time for #{issue_key}..."
-        started_at = Time.parse(options.fetch(:started_at, Time.now.to_s))
-        puts started_at
+        puts time_started_at
+        puts
+        puts "Press CTRL-P to pause/restart the timer"
+        puts "Press CTRL-C when you have finished to track the time"
+
+        reader = TTY::Reader.new
+        reader.on(:keyctrl_p) do
+          if time_paused_at == 0
+            time_paused_at = Time.now
+            partial_time_spent = partial_time_spent + (time_paused_at - time_restarted_at)
+
+            time_unit =
+                if (partial_time_spent / 60).floor == 1
+                  "minute"
+                else
+                  "minutes"
+                end
 
-        begin
-          loop do
-            sleep 1
+            puts
+            puts "Timer has been paused with #{(partial_time_spent / 60).floor} #{time_unit} logged"
+          else
+            time_paused_at = 0
+            time_restarted_at = Time.now
+            puts
+            puts "Timer has been restarted"
           end
+        end
+
+        loop do
+          reader.read_line("")
         rescue Interrupt
-          time_spent = (Time.now - started_at).floor
+          total_time_spent = (partial_time_spent + (Time.now - time_restarted_at)).floor
```

Comment:
Why flooring? Every variable is an integer/long here

Reply:
With the new commit all the floor are gone (see previous comment)